websocket.go

```diff
@@ -609,8 +620,63 @@ func (c *wsConn) tryReconnect(ctx context.Context) bool {
 	return true
 }
 
+func (c *wsConn) readFrame(ctx context.Context, r io.Reader) {
+	// debug util - dump all messages to stderr
+	// r = io.TeeReader(r, os.Stderr)
+
+	// json.NewDecoder(r).Decode would read the whole frame as well, so might as well do it
+	// with ReadAll which should be much faster
+	// use a autoResetReader in case the read takes a long time
+	buf, err := io.ReadAll(c.autoResetReader(r)) // todo buffer pool
```

Comment:
Ready to merge without buffer pool?

Reply:
Yeah, that's not really the point of the PR, and can be done separately